README.tfdoc.hcl

```
@@ -0,0 +1,1265 @@
+        description = <<-END
+          Name of the SSL Policy for the listener. Required if protocol is
+          `HTTPS` or `TLS`.
+        END
+      }
+
+      variable "additional_certificates_arns" {
+        type        = set(string)
+        default     = []
+        description = <<-END
+          The ARNs of the additional certificates to attach to the listener.
+        END
+      }
+
+      variable "default_action" {
+        type = object(default_action)
+        default = {
+          fixed_response = {
+            content_type = "plain/text"
+            message_body = "Nothing to see here!"
+            status_code  = 418
+          }
+        }
+        description = <<-END
+          Configuration block for default actions.
+        END
+
+        attribute "type" {
+          required    = true
+          type        = string
+          description = <<-END
+            Type of routing action. Valid values are `forward`, `redirect`,
+            `fixed-response`, `authenticate-cognito` and `authenticate-oidc`.
+          END
+        }
+
+        attribute "target_group_arn" {
+          type        = string
+          description = <<-END
+            ARN of the Target Group to which to route traffic. Specify only if
+            `type` is `forward` and you want to route to a single target group.
+            To route to one or more target groups, use a `forward` block instead.
+          END
+        }
+
+        attribute "order" {
+          type        = number
+          description = <<-END
+            Order for the action. This value is required for rules with multiple
+            actions. The action with the lowest value for order is performed
+            first. Valid values are between `1` and `50000`.
+          END
+        }
+
+        attribute "authenticate_cognito" {
+          type        = object(authenticate_cognito)
+          default     = []
+          description = <<-END
+            Configuration block for using Amazon Cognito to authenticate users.
+            Specify only when `type` is `authenticate-cognito`.
+          END
+
+          attribute "user_pool_arn" {
+            required    = true
+            type        = string
+            description = <<-END
+              ARN of the Cognito user pool.
+            END
+          }
+
+          attribute "user_pool_domain" {
+            required    = true
+            type        = string
+            description = <<-END
+              Domain prefix or fully-qualified domain name of the Cognito user pool.
+            END
+          }
+
+          attribute "user_pool_client_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              Domain prefix or fully-qualified domain name of the Cognito user
+              pool.
+            END
+          }
+
+          attribute "authentication_request_extra_params" {
+            type        = map(string)
```

```suggestion
            type        = list(authentication_request_extra_param)
```

This is a list of objects afaics

---

https://github.com/mineiros-io/terraform-aws-lb-listener/blob/main/variables.tf#L81